### DISCUSSION TITLE
What determines if "develop" generates files into .cache/page-ssr/routes?

### DISCUSSION THREAD

**Discussion by jonahark** [gatsbyjs/gatsby#36581] [Help]
We have 2 different behaviors in develop. 
1. Running in mono repo with our theme package --> .cache/page-ssr/routes is empty
2. Running in a separate repo with theme installed in node_modules --> .cache/page-ssr/routes contains a bundle for each page

Development works in both cases, but when developing in a separate repo the disk requirements increase exponentially. 4 files are created for each page. This has knock-on effects for the size of the webpack cache, start-up time, etc.

We don't use SSR at all, so we aren't actually expecting anything to be in this folder. 

Does anyone have suggestions on what may be causing this behavior?

**Comment by marvinjude** [comment]
Hi @jonahark,

gatsby creates bundles + maps files for page template and Head API in `.cache/page-ssr/routes` and this is used for HTML generation(So that gets created wether or not you're using SSR in your site. More details [here](https://www.gatsbyjs.com/docs/html-generation/#build-htmljs)). The only case where  we don't do this in develop is when you're in  [DEV_SSR mode ](https://www.gatsbyjs.com/docs/reference/release-notes/v2.30/#server-side-rendering-ssr-in-development)

**Reply by jonahark** [reply]
Thanks @marvinjude for the quick response!

Good to know what the expected behavior is.

Any thoughts on why the files aren't generated when site and theme are in the same monorepo?
![image](https://user-images.githubusercontent.com/114133/189339353-4478607e-1efb-4f4c-9a40-c7341c6901bf.png)

We don't have any flags for SSR. We're using the exact same site repo.
